src/mame/heathkit/tlb.cpp

```diff
@@ -0,0 +1,688 @@
+	ROM_LOAD( "2716_444-37_h19keyb.bin", 0x0000, 0x0800, CRC(5c3e6972) SHA1(df49ce64ae48652346a91648c58178a34fb37d3c))
+ROM_END
+
+ROM_START( super19 )
+	// Super H19 ROM
+	ROM_REGION( 0x2000, "maincpu", ROMREGION_ERASEFF )
+	ROM_LOAD( "2732_super19_h447.bin", 0x0000, 0x1000, CRC(6c51aaa6) SHA1(5e368b39fe2f1af44a905dc474663198ab630117))
+
+	ROM_REGION( 0x0800, "chargen", 0 )
+	// Original font dump
+	ROM_LOAD( "2716_444-29_h19font.bin", 0x0000, 0x0800, CRC(d595ac1d) SHA1(130fb4ea8754106340c318592eec2d8a0deaf3d0))
+
+	ROM_REGION( 0x1000, "keyboard", 0 )
+	// Original dump
+	ROM_LOAD( "2716_444-37_h19keyb.bin", 0x0000, 0x0800, CRC(5c3e6972) SHA1(df49ce64ae48652346a91648c58178a34fb37d3c))
+ROM_END
+
+ROM_START( watz19 )
+	// Watzman ROM
+	ROM_REGION( 0x2000, "maincpu", ROMREGION_ERASEFF )
+	ROM_LOAD( "watzman.bin", 0x0000, 0x1000, CRC(8168b6dc) SHA1(bfaebb9d766edbe545d24bc2b6630be4f3aa0ce9))
+
+	ROM_REGION( 0x0800, "chargen", 0 )
+	// Original font dump
+	ROM_LOAD( "2716_444-29_h19font.bin", 0x0000, 0x0800, CRC(d595ac1d) SHA1(130fb4ea8754106340c318592eec2d8a0deaf3d0))
+	// Watzman keyboard
+	ROM_REGION( 0x1000, "keyboard", 0 )
+	ROM_LOAD( "keybd.bin", 0x0000, 0x0800, CRC(58dc8217) SHA1(1b23705290bdf9fc6342065c6a528c04bff67b13))
+ROM_END
+
+ROM_START( ultra19 )
+	// ULTRA ROM
+	ROM_REGION( 0x2000, "maincpu", ROMREGION_ERASEFF )
+	ROM_LOAD( "2532_h19_ultra_firmware.bin", 0x0000, 0x1000, CRC(8ad4cdb4) SHA1(d6e1fc37a1f52abfce5e9adb1819e0030bed1df3))
+
+	ROM_REGION( 0x0800, "chargen", 0 )
+	// Original font dump
+	ROM_LOAD( "2716_444-29_h19font.bin", 0x0000, 0x0800, CRC(d595ac1d) SHA1(130fb4ea8754106340c318592eec2d8a0deaf3d0))
+	// Watzman keyboard
+	ROM_REGION( 0x1000, "keyboard", 0 )
+	ROM_LOAD( "2716_h19_ultra_keyboard.bin", 0x0000, 0x0800, CRC(76130c92) SHA1(ca39c602af48505139d2750a084b5f8f0e662ff7))
+#endif
+
+ROM_END
+
+
+ioport_constructor heath_tlb_device::device_input_ports() const
+{
+	return INPUT_PORTS_NAME(tlb);
+}
+
+const tiny_rom_entry *heath_tlb_device::device_rom_region() const
+{
+	return ROM_NAME(h19);
+}
+
+void heath_tlb_device::serial_out_b(uint8_t data)
+{
+	m_write_sd(data);
+}
+
+WRITE_LINE_MEMBER(heath_tlb_device::cb1_w)
+{
+	m_ace->rx_w(state);
+}
+
+void heath_tlb_device::device_add_mconfig(machine_config &config)
+{
+	// basic machine hardware
+	Z80(config, m_maincpu, H19_CLOCK); // From schematics
+	m_maincpu->set_addrmap(AS_PROGRAM, &heath_tlb_device::mem_map);
+	m_maincpu->set_addrmap(AS_IO, &heath_tlb_device::io_map);
+
+	// video hardware
+	// TODO: make configurable, Heath offered 2 different CRTs - White, Green
+	screen_device &screen(SCREEN(config, "screen", SCREEN_TYPE_RASTER, rgb_t::green()));
+	screen.set_refresh_hz(60);                         // TODO- this is adjustable by dipswitch.
+	screen.set_vblank_time(ATTOSECONDS_IN_USEC(2500)); /* not accurate */
+	screen.set_screen_update("crtc", FUNC(mc6845_device::screen_update));
+	screen.set_size(640, 250);
+	screen.set_visarea(0, 640 - 1, 0, 250 - 1);
```

Comment:
If the screen is controlled by the CRTC, then the refresh rate etc. will be updated automatically when the software reprograms the CRTC’s screen control registers.  You should still use `set_raw` to set the default screen parameters to the most common or default parameters programmed into the CRTC.